### ISSUE TITLE
Selecting Arabic subtitles causes the player to stop functioning

### ISSUE THREAD

**Issue by atabakselahi** [xbmc/xbmc#24874]
### Describe the bug
Here is a clear and concise description of what the problem is:

The movie plays smoothly with English subtitles; however, switching to Arabic subtitles causes the player to halt and exit after a few seconds.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

## Possible Fix

Selecting Arabic subtitles causes the player to halt and exit after a few seconds.

### To Reproduce

1. 
2.
3.

### Debuglog

The debuglog can be found here:

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [x] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version:

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by CastagnaIT** [comment]
please fullfill all template parts,
as it is it is not enough to investigate since to me arabic subtitle works
most important also provide a way to reproduce the problem, so a sample video or another way

**Comment by atabakselahi** [comment]
This error occurs on Kodi 21.0 “Omega” RC1 running on a Sony TV (Android TV). I’ve attached a sample video and subtitles . When you choose any external subtitle, initially I thought it was happening only with Arabic/Farsi subtitles. However, I noticed that it occurs even when I choose English subtitles. Interestingly, when I embedded the SRT file using mkvtoolnix into the MKV file, Kodi plays the file correctly. The link to file is: 

https://drive.google.com/file/d/10c8CYsu7m9ij6M0DAeLHRKRGOrvlGGWR/view?usp=drive_link

**Comment by neo1973** [comment]
Also no issues here on regular Linux with the provided example:
![grafik](https://github.com/xbmc/xbmc/assets/3134317/5d712eba-de54-4bda-9604-232b64da46ea)

**Comment by CastagnaIT** [comment]
i also tried on Android 64 and Windows and works as expected
since you havent provide the debug log there is nothing more to do

**Comment by atabakselahi** [comment]
Here is the log file for your refrence.
[kodi_subtitile_exit_palyer_error.log](https://github.com/xbmc/xbmc/files/15140703/kodi_subtitile_exit_palyer_error.log)

**Comment by atabakselahi** [comment]
> Here is the log file for your reference.

Here is the log file for your reference.
[kodi_subtitile_exit_palyer_error.log](https://github.com/xbmc/xbmc/files/15156756/kodi_subtitile_exit_palyer_error.log)

**Comment by CastagnaIT** [comment]
at least from here the problem seem not related to subtitles
```
2024-04-27 23:34:48.171 T:7893    error <general>: Read - Error( -1, 32, Broken pipe )
2024-04-27 23:34:48.171 T:7893  warning <general>: underflow: Error reading file - assuming eof
2024-04-27 23:34:48.171 T:7893  warning <general>: CFileCache::Process - <smb://RT-AX86S-6878/My_Passport/Movies/Tar 2022/tar.2022.2160p.uhd.bluray.x265-b0mbardiers.mkv> source read returned 0! Will retry
2024-04-27 23:34:48.172 T:23679    info <general>: CAESinkAUDIOTRACK::Initializing with: m_sampleRate: 192000 format: AE_FMT_S16LE (AE) method: IEC (PT) stream-type: STREAM_TYPE_TRUEHD min_buffer_size: 393472 m_frames: 6148 m_frameSize: 16 channels: 8
2024-04-27 23:34:50.172 T:7893    error <general>: Read - Error( -1, 32, Broken pipe )
2024-04-27 23:34:50.172 T:7893  warning <general>: underflow: Error reading file - assuming eof
2024-04-27 23:34:50.172 T:7893  warning <general>: CFileCache::Process - <smb://RT-AX86S-6878/My_Passport/Movies/Tar 2022/tar.2022.2160p.uhd.bluray.x265-b0mbardiers.mkv> source read returned 0! Will retry
2024-04-27 23:34:52.173 T:7893    error <general>: Read - Error( -1, 32, Broken pipe )
2024-04-27 23:34:52.173 T:7893  warning <general>: underflow: Error reading file - assuming eof
2024-04-27 23:34:52.173 T:7893  warning <general>: CFileCache::Process - <smb://RT-AX86S-6878/My_Passport/Movies/Tar 2022/tar.2022.2160p.uhd.bluray.x265-b0mbardiers.mkv> source read returned 0! Will retry
2024-04-27 23:34:54.174 T:7893    error <general>: Read - Error( -1, 32, Broken pipe )
2024-04-27 23:34:54.174 T:7893  warning <general>: underflow: Error reading file - assuming eof
2024-04-27 23:34:54.174 T:7893    error <general>: CFileCache::Process - <smb://RT-AX86S-6878/My_Passport/Movies/Tar 2022/tar.2022.2160p.uhd.bluray.x265-b0mbardiers.mkv> source read didn't return any data before eof!
2024-04-27 23:34:54.220 T:7890    error <general>: ffmpeg[0xd09c7c30]: [matroska,webm] File ended prematurely
2024-04-27 23:35:00.187 T:7899     info <general>: CVideoPlayerAudio::Process - stream stalled
2024-04-27 23:35:00.231 T:7890     info <general>: Process - eof reading from demuxer
```

if it is not a corrupted video file,
imo could be related to samba cache but im not expert on this
on kodi 21 Settings->Services->SMB
try change the Chunk size to some higher number

**Comment by scott967** [comment]
Have seen reports like this and seems SMB related in that client is using SMBv1 with chunk size 128 when v1 only supports up to 64.  Best fix is to migrate to SMBv3 but changing chunk size to 64 kb in settings has been reported to work.

**Comment by atabakselahi** [comment]
Thank you for your response and the information provided. However, these errors do not cause the player to exit. Instead, Kodi exits when I select the .srt file. I am able to play the video from the SMB without any issues when not using external subtitles. The error related to this issue is below, and it seems to be associated with ‘CMediaCodec’.”:

2024-04-27 17:32:31.080 T:23652   error <general>: CMediaCodecVideoBuffer::ReleaseOutputBuffer error in render(false)
2024-04-27 17:32:36.402 T:23737    info <general>: Skipped 2 duplicate messages..
2024-04-27 17:32:36.401 T:23737 warning <general>: OutputPicture - timeout waiting for buffer
2024-04-27 17:32:36.445 T:23737   error <general>: CMediaCodecVideoBuffer::ReleaseOutputBuffer error in render(false)
2024-04-27 17:32:36.546 T:23738    info <general>: Skipped 3 duplicate messages..
2024-04-27 17:32:36.546 T:23738    info <general>: CAEStreamParser::SyncDTS - dtsHD MA stream detected (8 channels, 48000Hz, 16bit BE, period: 8192, syncword: 0x41a29547, target rate: 0x18, framesize 5292))
2024-04-27 17:32:40.347 T:23737   error <general>: CMediaCodecVideoBuffer::ReleaseOutputBuffer error in render(false)
2024-04-27 17:32:40.358 T:23738    info <general>: Skipped 3 duplicate messages..
2024-04-27 17:32:40.358 T:23738    info <general>: CAEStreamParser::SyncDTS - dtsHD MA stream detected (8 channels, 48000Hz, 16bit BE, period: 8192, syncword: 0x41a29547, target rate: 0x18, framesize 7184))
2024-04-27 17:33:03.600 T:23737 warning <general>: OutputPicture - timeout waiting for buffer
2024-04-27 17:33:03.604 T:23652   error <general>: CMediaCodecVideoBuffer::ReleaseOutputBuffer error in render(true)
2024-04-27 17:33:03.640 T:23737   error <general>: CMediaCodecVideoBuffer::ReleaseOutputBuffer error in render(false)
2024-04-27 17:33:03.702 T:23728    info <general>: Skipped 1 duplicate messages..
2024-04-27 17:33:03.702 T:23728    info <general>: Creating InputStream
2024-04-27 17:33:03.821 T:23728    info <general>: Creating Demuxer
2024-04-27 17:33:03.983 T:23784    info <general>: CDVDVideoCodecFFmpeg::Open() Using codec: MJPEG (Motion JPEG)
2024-04-27 17:33:04.415 T:23728    info <general>: Opening stream: 0 source: 256
2024-04-27 17:33:04.415 T:23728    info <general>: Creating video codec with codec id: 173
2024-04-27 17:33:04.416 T:23728   error <general>: CDVDVideoCodecAndroidMediaCodec::Open - InstanceGuard locke

**Comment by neo1973** [comment]
This has been fixed by #25128.
